packages/server/api/webscrape.py

```diff
@@ -3,32 +3,63 @@
 import re
 from urllib import parse
 from bs4 import BeautifulSoup
+import json
 
 class handler(BaseHTTPRequestHandler):
   whitelisted_url_regexes = [
     re.compile('^https:\/\/www\.bonappetit\.com')
   ]
 
+  def error_handler(error_string, error_code = 500):
+    self.send_response(error_code)
+    self.send_header('Content-type', 'text/plain')
+    self.end_headers()
+    self.wfile.write(errror_string.encode())
+
   def do_GET(self):
     url = parse.parse_qs(parse.urlsplit(self.path).query)['url'][0]
     url_in_whitelist = any(regex.match(url) for regex in self.whitelisted_url_regexes)
     
     if(not url_in_whitelist):
-      self.send_response(500)
-      self.send_header('Content-type', 'text/plain')
-      self.end_headers()
-      self.wfile.write('Unsupported recipe site'.encode())
+      error_handler('URL not supported')
+      return
+    
+    recipe_object = {
+      'servings': '',
+      'prepTime': '',
+      'cookTime': '',
+      'ingredients': '',
+      'steps': '',
+      'links': '',
+      'name': ''
+    }
+
+    try: 
+      file_pointer = request.urlopen(url)
+      html_bytes = file_pointer.read()
+      html_doc = html_bytes.decode('utf8')
+      file_pointer.close()
+      html_parsed = BeautifulSoup(html_doc, 'html.parser')
+    except:
+      error_handler('HTML could not be parsed')
       return
 
-    file_pointer = request.urlopen(url)
-    html_bytes = file_pointer.read()
-    html_doc = html_bytes.decode('utf8')
-    file_pointer.close()
-    html_parsed = BeautifulSoup(html_doc, 'html.parser')
-    ingredients = html_parsed.find('div', 'ingredients')
+    try: 
+      recipe_object['name'] = html_parsed.find('a', 'top-anchor').text
+      recipe_object['servings'] = html_parsed.find('span', 'recipe__header__servings recipe__header__servings--basically').span.text
+      ingredients = html_parsed.find_all('div', 'ingredients__text')
+      ingredients_array = list(map(lambda ingredient : { 'name': ingredient.text}, ingredients))
```

Comment:
I think the more pythonic way of writing this would be
```suggestion
      ingredients_array = list([{ 'name': ingredient.text } for ingredient in ingredients])
```

Reply:
Arent map functions more efficient than for loops?

Reply:
Syntax is clean though, just wondering haha

Reply:
Not in python https://leadsift.com/loop-map-list-comprehension/

Reply:
Ok sounds good, I like the loop syntax more